src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    int size;
+    Node<T> first;
+    Node<T> last;
+
+    private static class Node<T> {
+        T item;
+        Node<T> next;
+        Node<T> prev;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        linkLast(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, nodeAt(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        Object[] arr = list.toArray();
+        if (arr.length == 0) {
+            return false;
+        }
+        for (T i:list) {
+            linkLast(i);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return nodeAt(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = nodeAt(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = nodeAt(index);
+        unlink(node);
+        return node.item;
     }
 
     @Override
     public boolean remove(T t) {
+        for (Node<T> x = first; x != null; x = x.next) {
+            if (x.item == t || x.item != null && x.item.equals(t)) {
+                unlink(x);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public void indexCheck(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    public void linkLast(T value) {
+        final Node<T> l = last;
```

Comment:
1. let's use the informative name of the variables in all methods
2. do we need final here?